File: Test/2020_techcamp/pro01.js
const countNumber = (inputArr) => {
    let [tmp, ...arr] = inputArr
    let count = 1
    let countArr = []
    for (const a of arr) {
        if (tmp === a) {
            count += 1
        }
        else {
            countArr.push(count)
            tmp = a
            count = 1
        }
    }
    countArr.push(count)
    return countArr
}

const solution = (inputArr) => {
    let count = 0
    while (!(inputArr.length === 1 && inputArr[0] === 1)) {
        inputArr = countNumber(inputArr)        
        count +=1
    }
    return count
}


console.log(solution([1, 1, 3, 3, 2, 2, 4, 5, 1, 1, 1, 3, 3, 3]	))

File: Test/2020_kakao/pro01.py
def delete_wrong_char(string):
  result = ''
  for s in string:
    if s.isalnum() or s in ['-', '_', '.']:
      result += s
  return result

def delete_double_dot(string):
  result = ''
  tmp = False
  for s in string:
    if tmp and s == '.': continue
    result += s
    tmp = (s == '.')
  return result

def delete_endpoint_dot(string):
  start = 0
  end = len(string)
  if string[start] == '.':
    start += 1
  if string[end - 1] == '.':
    end -= 1
  return string[start : end]

def add_char(count, char, string):
  string = string + (count * char)
  return string


def solution(new_id):
  new_id = new_id.lower() # 소문자로 치환
  new_id = delete_wrong_char(new_id) # 불필요한 문자 제거
  new_id = delete_double_dot(new_id) # 연속된 점 제거
  new_id = delete_endpoint_dot(new_id) # 양끝 점 제거
  if not new_id:
    new_id = add_char(1, 'a', new_id) # 빈 문자열이면 a 더하기
  new_id = new_id[0:15] # 15자까지 자르기
  new_id = delete_endpoint_dot(new_id) # 양끝 점 제거
  if len(new_id) <= 2: 
    new_id = add_char(3 - len(new_id), new_id[-1], new_id) # 지정된 길이만큼 마지막 문자 더하기
  return new_id

solution("...!@BaT#*..y.abcdefghijklm."	)
solution("=.="	)